hubble-proxy/cmd/context/context.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package context
+
+import (
+	"io"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+// Context is a context for commands. It holds a reference to the command
+// configuration, a logger and I/O streams.
+type Context struct {
```

Comment:
im' guessing this is probably something ported from cilium, but `context.Context` has a way to pass objects around as well, without the need for a completely separate function signature to pass some additional context structure.

https://golang.org/pkg/context/#WithValue

Reply:
It is not something ported from cilium. The main purpose of this command context is to avoid global vars for the logger and config object to allow for easier testing.

If you insist, I can embed this information into  `context.Context` as you suggest but this will come at the cost of more boilerplate code (to handle `interface{}` type conversion, defining own type for keys as it is a requirement, etc) and decreased visibility (it's not obvious by reading a function signature that the context parameter is expected to carry special values).

Reply:
Context bag is meant for things like rpc/http servers that have to do things over and over. The cmd is bootstrapped once, why do we need context at all?